src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java

```diff
@@ -329,15 +317,27 @@ private InitValueView getInitValue(MethodTransformerContext context, InitCodeNod
         initValue.initialValue(
             context.getTypeTable().getZeroValueAndSaveNicknameFor(item.getType()));
       }
-
       return initValue.build();
     }
   }
 
   private static List<String> getFormatFunctionArgs(
-      List<String> varList, InitValueConfig initValueConfig) {
+      MethodTransformerContext context, Field field, InitValueConfig initValueConfig) {
     List<String> formatFunctionArgs = new ArrayList<>();
-    for (String entityName : varList) {
+
+    Set<String> entityNames;
+    boolean shouldUseCollectionConfigEntityNames =
+        !context.getFeatureConfig().useResourceNameFormatOption(field)
+            || initValueConfig.hasFormattingConfigInitialValues();
+    if (shouldUseCollectionConfigEntityNames) {
```

Comment:
I found where the "_id" suffix is coming from:
https://github.com/googleapis/toolkit/blob/master/src/test/java/com/google/api/codegen/testdata/library.proto#L30

This is a problem that will be fixed once the resource name configuration moves out of the proto and into the gapic config, although it was a mistake to allow this difference in the first place - my original PR should have checked for this inconsistency.

In this PR, which do you think is preferable - the changes you have here, or switching shelf_id->shelf in the proto, or both?